File: models/OfferType.js
const utils = require('../utils/utils');

module.exports = {
        fields: (key) => (
            {
                label: `Особый тип товара:  * `MEDICINE` — лекарства. * `BOOK` — бумажные и электронные книги. * `AUDIOBOOK` — аудиокниги. * `ARTIST_TITLE` — музыкальная и видеопродукция. * `ON_DEMAND` — товары на заказ.  {% note info \"Если ваш товар — книга\" %}  Укажите год издания в характеристиках товара. [Подробнее о параметре](../../reference/business-assortment/updateOfferMappings.md#offerparamdto)  {% endnote %}  - [${key.replaceAll('__', '.')}]`,
                choices: [
                    'DEFAULT',
                    'MEDICINE',
                    'BOOK',
                    'AUDIOBOOK',
                    'ARTIST_TITLE',
                    'ON_DEMAND',
                ],
            }
        )
    }

File: src/main/scala/org/openapitools/models/OfferType.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import ympa_scala_finch_server._

/**
 * Особый тип товара:  * `MEDICINE` — лекарства. * `BOOK` — бумажные и электронные книги. * `AUDIOBOOK` — аудиокниги. * `ARTIST_TITLE` — музыкальная и видеопродукция. * `ON_DEMAND` — товары на заказ.  {% note info \"Если ваш товар — книга\" %}  Укажите год издания в характеристиках товара. [Подробнее о параметре](../../reference/business-assortment/updateOfferMappings.md#offerparamdto)  {% endnote %} 
 */
case class OfferType()

object OfferType {
    /**
     * Creates the codec for converting OfferType from and to JSON.
     */
    implicit val decoder: Decoder[OfferType] = deriveDecoder
    implicit val encoder: ObjectEncoder[OfferType] = deriveEncoder
}
